scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/factories/model/DefaultModelJobValidatorFactory.java

```diff
@@ -0,0 +1,391 @@
+ * TaskFlowJob validator which validates variables against model definition
+ */
+public class DefaultModelJobValidatorFactory implements JobValidatorFactory {
+
+    protected static final Logger logger = Logger.getLogger(DefaultModelJobValidatorFactory.class);
+
+    public static final String INTEGER_MODEL = "INTEGER_MODEL";
+
+    public static final String LONG_MODEL = "LONG";
+
+    public static final String DOUBLE_MODEL = "DOUBLE";
+
+    public static final String BOOLEAN_MODEL = "BOOLEAN";
+
+    public static final String FLOAT_MODEL = "FLOAT";
+
+    public static final String SHORT_MODEL = "SHORT";
+
+    public static final String URL_MODEL = "URL";
+
+    public static final String URI_MODEL = "URI";
+
+    public static final String DATETIME_MODEL = "DATETIME";
+
+    public static final String LIST_MODEL = "LIST";
+
+    public static final String REGEXP_MODEL = "REGEXP";
+
+    public static final String MODEL_FROM_URL_MODEL = "MODEL_FROM_URL";
+
+    public DefaultModelJobValidatorFactory() {
+        // empty
+    }
+
+    @Override
+    public void validateJob(File jobFile) throws JobCreationException {
+        // validate any job
+    }
+
+    @Override
+    public void validateJob(TaskFlowJob job) throws JobCreationException {
+        for (Task task : job.getTasks()) {
+            for (TaskVariable variable : task.getVariables().values()) {
+                checkVariableFormat(task, variable);
+            }
+        }
+    }
+
+    public void checkVariableFormat(Task task, TaskVariable variable) throws JobCreationException {
+        if (!Strings.isNullOrEmpty(variable.getModel())) {
+            String model = variable.getModel().trim();
+            String upperCaseModel = model.toUpperCase();
+            switch (upperCaseModel) {
+                case INTEGER_MODEL:
+                    checkValueOf(task, variable, model, new Integer(0));
+                    return;
+                case LONG_MODEL:
+                    checkValueOf(task, variable, model, new Long(0));
+                    return;
+                case DOUBLE_MODEL:
+                    checkValueOf(task, variable, model, new Double(0));
+                    return;
+                case BOOLEAN_MODEL:
+                    checkBoolean(task, variable, model);
+                    return;
+                case FLOAT_MODEL:
+                    checkValueOf(task, variable, model, new Float(0));
+                    return;
+                case SHORT_MODEL:
+                    checkValueOf(task, variable, model, new Short((short) 0));
+                    return;
+                case URL_MODEL:
+                    checkURL(task, variable, model);
+                    return;
+                case URI_MODEL:
+                    checkURI(task, variable, model);
+                    return;
+                default:
+                    // not one of the above models
+            }
+
+            if (upperCaseModel.startsWith(INTEGER_MODEL)) {
```

Comment:
This second set of conditions is for checking range validity. Please split this very long method into 2 methods with a clear purpose.